New VFX System

SantyWang opened the pull request:
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

<!-- Note: Makes sure these boxes are checked before submitting your PR - thank you!
- [ ] Your pull request title is using English, it's precise and appropriate.
- [ ] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [ ] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- [ ] Document new code with comments in source code based on API docs
- [ ] Make sure any runtime log information in `log` , `error` or `new Error('')` has been moved into `EngineErrorMap.md` with an ID, and use `logID(id)` or `new Error(getError(id))` instead.
- To official teams:
  - [ ] Check that your PR is following our [guides](https://github.com/cocos/3d-tasks/blob/master/workflows/readme.md)
-->

shrinktofit commented on cocos/core/curves/curve.ts:
```diff
@@ -401,14 +402,18 @@ export class RealCurve extends KeyframeCurve<RealKeyframeValue> {
 
         const iNext = ~index;
-        assertIsTrue(iNext !== 0 && iNext !== nFrames && nFrames > 1);
+        if (DEBUG) {
```
This method can't be shaked?

SantyWang committed:
refine data-set interface

zxx43 committed:
add dynamic buffer (#32)

* add dynamic buffer

* modify dynamic buffer

* move functions to vfx-manager

* update buffer once

xubing0906 committed:
add setMaterialProperty function for skyboxInfo (#15284)

2youyou2 committed:
fixed fsr param not saved (#15301)

lealzhan committed:
[fix] physx misuses bullet variable (#15255)

SantyWang committed:
refine dynamic buffer

2youyou2 committed:
fixed fsr input size (#15304)

troublemaker52025 committed:
performance optimization for onMacroPatchesStateChanged function (#15279)

SantyWang committed:
refine renderer

troublemaker52025 committed:
The globalStandardSkin setting is placed on the meshRenderer component (#15274)

shrinktofit committed:
Marionette/PoseGraph: allow specify start time for PoseNodePlayMotion (#15306)

qiuguohua committed:
Fixed safe areas requires dpr (#15275)

knoxHuang committed:
Fix the problem that adding asset to the search bar is invalid (#15283)

shrinktofit committed:
Marionette/StateMachine: fix state motion time binding (#15308)

shrinktofit committed:
Animation graph: Fix visitAnimationClips() editor API (#15197)

PatriceJiang committed:
add proguard-configs for android/release (#15315)

shrinktofit committed:
Marionette: add editor method to copy state machine as pose graph node (#15278)

* Marionette: add editor method to copy state machine as pose graph node

* Fix copy

* Fix Clone

xubing0906 committed:
fix planar reflection update submodel macro patches (#15316)

zxx43 committed:
add ribbon renderer (#33)

SantyWang committed:
refine shader

SantyWang committed:
Merge branch 'new-particle' of github.com:holycanvas/engine into v3.6.2-particle

fqamic committed:
feat: sync android template, update gradle 8.0.2 (#15318)

yanOO1497 committed:
Revert "[Optimize] Optimize missing asset error reporting (#15154)" (#15317)

This reverts commit ed1cf6b3d987f830c679f575e76c7bd84d74bdc0.

yiwenxue committed:
fix code style (#15310)

* fix code style

* rm getter

LinYunMo committed:
HD processing of ttf mode labels (#15206)

* Adding ttf is affected by dpr

* Adding Char mode is affected by dpr

* move dpr to style

* rebase & add max font size

* name change

* remove use less

* misc

SantyWang committed:
refine shader

SantyWang committed:
refine shader

yiwenxue committed:
[fix] assign physics system to PhysicsSystem._instance, before using it (#15320)

troublemaker52025 committed:
Fix skin with post-process pipeline (#15327)

arsen2010 committed:
optimize inspector interface (#15325)

* optimize inspector interface

* adjust inspector group style

yiwenxue committed:
[fix] assign physics system to PhysicsSystem._instance, before using it (#15331)

PatriceJiang committed:
Android support CJK app name (#15326)

PPpro committed:
ban wasm support on bytedance platoform (#15332)

jk20012001 committed:
fix zeno vat blend and instancing non-uniform scaling (#15335)

oahc09 committed:
fix:xr template proguard add dontwarn for gvr sdk when using gradle v8.x (#15323)

naturalch committed:
optimizing LOD mesh trianglecount display (#15336)

xubing0906 committed:
fix terrain error in custom pipeline (#15339)

oahc09 committed:
fix:xr right eye render failed when using custom render pipeline (#15319)

yiwenxue committed:
Fix texture mipmaps with compression (#15277)

SantyWang committed:
refine shader

dogodo-cc committed:
feat: inspector/texture/ Comparison of the differences between atlas and useData regarding the configuration of filter (#15203)

* feat: inspector/texture/ Comparison of the differences between atlas and useData regarding the configuration of filter

* feat: update color

troublemaker52025 committed:
fix skin pipeline scene data crash with native scene (#15341)

naturalch committed:
Revert "save settings to pac (#14917)" (#15344)

This reverts commit 2bb1c366b3672553ea5ddf6366a3f6f4ca087e5d.

LinYunMo committed:
fix font scale limit bug (#15342)

* fix font scale limit bug

* name change

pandamicro committed:
Fix issues in Node.updateWorldTransform and UITransform.getBoundingBoxTo[World] (#15273)

* Fix transform update issue when only rotation is set to root node

* Fix UITransform.getBoundingBoxTo and UITransform.getBoundingBoxToWorld

* Improve API and test

* Fix native Node::updateWorldTransform

* Fix UITransform.getBoundingBoxTo

* Fix getBoundingBoxTo by ensuring one time rect transform

* Ignore zero sized rect for bounding box calculation

* Improve API doc

shrinktofit committed:
Marionette/Stash: fix error if reenter() the stash but updated in later (#15337)

* Marionette/Stash: fix error if reenter() the stash but updated in later

* Optimize stash evaluate state

arsen2010 committed:
optimize inspector editing asset interface (#15343)

* optimize inspector editing asset interface

* fix animation panel time precision bug

* optimize editing interface of asset in inspector

jk20012001 committed:
fix hbao with fog and skin depth  (#15340)

naturalch committed:
fix apply current size to lod (#15338)

* fix apply current size to lod

* update

fqamic committed:
feat: xr sync android template, support CJK app name (#15346)

shrinktofit committed:
Marionette/PoseGraph: uniformly decide if a weight can be ignore (#15348)

LinYunMo committed:
fix bm font anchor bug (#15353)

moshuying committed:
[Bug] Fix native scene can't get UUID bug (#15347)

bofeng-song committed:
remove uneeded code & fix the issue of iOS not displaying in full screen. (#15352)

moshuying committed:
fix getUniform API bug (#15333)

Co-authored-by: James Chen <dumganhar@gmail.com>

novain1102 committed:
support Pacer to set specific frameRate without frame lost (#15350)

naturalch committed:
fix image type change error (#15351)

dogeFu committed:
feature : add EDITOR_NOT_IN_PREVIEW marco (#15354)

* feature : add marco for editor preview and refactor code with new marco

* refactor: replace GAME_VIEW with EDITOR_PREVIEW

* Revert "refactor: replace GAME_VIEW with EDITOR_PREVIEW"

This reverts commit e28953e9e7acbb49dc5bdc650b2ff65510d4b084.

* update cc config json

* refactor code

* remove uselese parentheses

* set ccglobal to be false

* skip global variable test case

troublemaker52025 committed:
Fix comment (#15358)

May2nd committed:
- fix memory leak on tiled resized (#15212)

Co-authored-by: Viet <>

shrinktofit committed:
Refactor animation graph event dispatching mechanism (#15330)

* Refactor animation graph event mechanism

* Fix test

* Update

* Revert

SantyWang committed:
fix editor preview error (#15290)

LinYunMo committed:
fix label auto Atlas bug (#15356)

yiwenxue committed:
Update external tag (#15361)

* [fix] add boundary to avoid physx assert error (#15321)

* sync external tag

yiwenxue committed:
fix bufferview size limit (#15265)

* fix bufferview size limit

* update

* update

* update error msg

* [fix] add boundary to avoid physx assert error (#15321)

* change validation method

xubing0906 committed:
reflection-probe-manager move to 3d module (#15359)

* move to 3d module

* export probetype

* fix error

arsen2010 committed:
[Bug] fix widget interface bug in inspector (#15364)

* fix widget interface bug

* fix image interface bug

* revert margin-right 4px in inspector

* optimize fbx interface in inspector

* change animation-graph integer variable  color

2youyou2 committed:
fixed editor cameras (#15365)

novain1102 committed:
fix setFrameRate unit-test (#15369)

lealzhan committed:
[opt] cct api refactor (#15360)

* [opt] cct api refactor

* tweaks

* tweak

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: c1485f4500e501c56632ec8d4f1fd1efea0c3202  (#15265) (#15362)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

yanOO1497 committed:
update spriteFrame meta version (#15380)

* update sprite version

* update

arsen2010 committed:
fix particle and fbx interface bug (#15376)

* fix widget interface bug

* fix fbx event-editor interface bug

* revert particle-3d interface bug in inspector

doommm committed:
Fix: button alignment of SpriteFrame asset (#15379)

star-e committed:
Develop pipeline merge (#15374)

jk20012001 committed:
fix post-process clamp sampling (#15382)

shrinktofit committed:
Marionette: fix animation blend 2d algorithm type is not setup correctly (#15377)

shrinktofit committed:
Marionette/StateMachine: fix any state won't match proc pose states (#15381)

LinYunMo committed:
limit font scale to 1 (#15367)

* limit font scale to 1

* to fix 1

* move

moshuying committed:
Update the type returned by get worldBounds (#15378)

star-e committed:
V3.8.0 pipeline (#15383)

bluesky013 committed:
Using the Vulkan backend is prohibited on Android devices running versions below Android 8 (APILevel <= 27). (#15386)

bluesky013 committed:
Using the Vulkan backend is prohibited on Android devices running versions below Android 8 (APILevel <= 27). (#15386)

bravetwo committed:
fix: adjust ar background binded texture sampler to fix limit on AREngine on GLES3 (#15389)

PPpro committed:
fix: bullet crash on Wechat iOS (#15384)

novain1102 committed:
support displayRatio for wechat firstScreen (#15391)

* support displayRatio for wechat firstScreen

* use first-screen.ejs to replace first-screen.js

PatriceJiang committed:
Add Android GPG keyboard & Mouse support (#15385)

* Add Android GPG keyboard & mouse support

PatriceJiang committed:
revert a6684b1 (#15393)

SantyWang committed:
fix V3.8.0 performance unexpected downgrade (#15392)

* fix performance downgrade

* changed to EDITOR_NOT_IN_PREVIEW

* Update object.ts

---------

Co-authored-by: Jare Guo <jareguo@users.noreply.github.com>

PPpro committed:
add FORCE_BANNING_BULLET_WASM constant (#15394)

* add FORCE_BANNING_BULLET_WASM constant

* optimize getImportObject

jk20012001 committed:
performance optimization for tone mapping pass and rename function (#15395)

zxx43 committed:
add ribbon shader (#34)

zhakesi committed:
spine wasm version (#15309)

* spine wasm version -fix conflict

* fix build and ci test -fix conflict

* fix for code review

* support binary format

* update exernal lib

* revert license

* fix type any

* update api docs

* fix api docs

star-e committed:
V3.8.0 pipeline (#15398)

troublemaker52025 committed:
Adding project macro definitions to the native engine (#15399)

SantyWang committed:
refactor parameter

SantyWang committed:
refactor parameter

SantyWang committed:
Merge branch 'new-particle' of github.com:holycanvas/engine into v3.6.2-particle

SantyWang committed:
refactor parameter

jk20012001 committed:
fix shader compatible for old version (#15407)

troublemaker52025 committed:
fix csm with custom pipeline (#15409)

shrinktofit committed:
Fix clip frame events are not fired in animation graph (#15405)

* Support clip frame events in animation graph

* Opt

* Opt

SantyWang committed:
refactor parameter

SantyWang committed:
refactor parameter

shrinktofit committed:
PoseGraph: more detailed stash visit editor API (#15401)

shrinktofit committed:
Clear unused logs (#15419)

yiwenxue committed:
Update constraint comments (#15422)

* [fix] add boundary to avoid physx assert error (#15321)

* update comments for physics constraint components

SantyWang committed:
refine code

GengineJS committed:
[Delete]Remove post-processing configuration items (#15413)

shrinktofit committed:
Marionette/PoseGraph: update i18n (#15420)

dogodo-cc committed:
fix: inspector/texture: fault-tolerant for parse atlas (#15414)

* fix: inspector/texture: fault-tolerant for parse atlas

* feat: if parentAssetList is not Array, hide the tip

* feat: delete hyphhyphen

shrinktofit committed:
Adjust code order to eliminate warning (#15418)

qiuguohua committed:
Update Harmonyos third-party library. (#15429)

PatriceJiang committed:
add validation for cMakeConfig values (#15432)

shrinktofit committed:
PoseGraph: fix PoseNodeApplyTransform i18n (#15400)

naturalch committed:
update change image type (#15439)

dogeFu committed:
feat：add clear method for AnimationGraphPartialPreviewer (#15441)

* feat : add clearVariables method for AnimationGraphPartialPreviewer

* clear motionRecords too

SantyWang committed:
refactor compilation mechanism

SantyWang committed:
add emitter-state

star-e committed:
V3.8.0 pipeline (#15428)

moshuying committed:
revert preview change (#15450)

shrinktofit committed:
AnimationGraph: renaming variable should consider PVNodeGetVariable (#15460)

arsen2010 committed:
optimize asset and node interface in inspector (#15459)

* optimize inspector padding interface

* clear somewhere ui-prop ui-section margin-top

* optimize asset and node interface in inspector

* adjust asset.css

star-e committed:
[V3.8.0 pipeline] add deferred pipeline (#15452)

arsen2010 committed:
optimize ui-section header margin-right (#15462)

* optimize ui-section header margin-right

* update

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: dc2f864b2199db766b73b41910fd1f64035b6b38  (#15452) (#15463)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

SantyWang committed:
refactor rand generator

PPpro committed:
support fetchBuffer in pal/wasm module (#15454)

LinYunMo committed:
fix android label shadow Pollution (#15470)

SantyWang committed:
use new rand instead

doommm committed:
[Optimize] improve: rendering logic of effect select options (#15446)

* improve: rendering logic of effect select options

* update: escape option label string

novain1102 committed:
fix pacer-unit-test (#15461)

PPpro committed:
move spine module out of wechatPlugin (#15468)

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: 938bfeca4a7bd083beb52e795821305dd9444e27  (#15462) (#15465)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

lealzhan committed:
[fix] add character controller doc links; replace onGround with isGrounded (#15447)

PPpro committed:
optimize: load spine asm.js module with js.mem && update spine (#15469)

* optimize: load spine asm.js module with js.mem

* update spine (#12)

* support debug renderer

* fix cached mode update color

* enable Batch mode

* fix engineInternal

* fix ESLint

---------

Co-authored-by: zhakesi <zhakesi@foxmail.com>

xubing0906 committed:
fix toon outline thickness issue with vulkan (#15458)

PPpro committed:
add CULL_ASM_JS_MODULE constant (#15467)

* add CULL_ASM_JS_MODULE constant

troublemaker52025 committed:
add skin JSB binding (#15464)

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: 9781724f14b84955bf9a0c1b4073b36d18e26921  (#15467) (#15476)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

PatriceJiang committed:
3.8 fix android gradle plugin version (#15417)

* bugfix: android gradle vesion

* rm applicationId

LinYunMo committed:
Fix Android text truncation problem (#15448)

* Fix Android text truncation problem

* fix calculation bias

* move

* limit android

* Revert "limit android "

This reverts commit 12dfcd0b3ed82be0cf3cf16340859bfd444166d8.

* Revert "Revert "limit android ""

This reverts commit 2aaf4162ada7850672c532098e530636e593642f.

* remove round

* fix

SantyWang committed:
support parameter-registry

zhakesi committed:
add spine wasm c++ code (#15482)

* add spine wasm c++ code

* fix code style

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: cbe85192076db9a033607ae6c08cb3977f64d3cf  (#15464) (#15477)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

pandamicro committed:
Update external-config.json for new spine lib (#15483)

arsen2010 committed:
optimize asset editing interface (#15481)

star-e committed:
'Custom' and 'Forward' pipeline is defaulted to PostProcessBuilder (#15466)

star-e committed:
V3.8.0 pipeline (#15487)

dogeFu committed:
fix : undo remove label component cause error in editor (#15475)

* fix : add onRestore implement for label component

* refactor code

novain1102 committed:
fix ios keyboard switch bug when use numberic (#15486)

yiwenxue committed:
[fix] physics hinge transform (#15442)

* [fix] add boundary to avoid physx assert error (#15321)

* fix native physics transform problem

arsen2010 committed:
fix fbx and widget bug in inspector (#15492)

* optimize fbx editing and asset panel scrollbar

* optimize effect tab interface

* fix widget resizeObserver loop limit exceeded

* fix interface twinkle when switch node selection

* fix bug of fbx preview editor did not dispatch change

* revert code

troublemaker52025 committed:
[bug] fix wild pointers (#15489)

* [bug] fix wild pointers

star-e committed:
sync generated code with 3.8.1 (#15490)

yiwenxue committed:
Fix/skinned mesh max joints (#15495)

* fix maxjoints estimation

jk20012001 committed:
fix hair effect tranparency pass and terrain cast shadow (#15494)

lealzhan committed:
[fix] increase box2d debug line width from 2 to 3 (#15484)

zhakesi committed:
fix jsb root function destroyModel named removeModel (#15497)

SantyWang committed:
support vfx parameter registry and binding parameter

hana-alice committed:
avoid unexpected glslang behavior (#15488)

lealzhan committed:
[fix] math Quaternion::fromMat3 (#15491)

* [fix] native math Quaternion::fromMat3 and Mat3::fromViewUp

oahc09 committed:
fix:spaces xr build template miss Log import (#15473)

lealzhan committed:
[fix][physics2d] increase builtin debug line width from 1 to 3 (#15500)

PPpro committed:
fix: engine init stucked by exception when load spine wasm (#15498)

LinYunMo committed:
Add get stats interface in profiler (#15474)

* add get stats interface in profiler

* change to getter

* change name

* misc

* use warn

xubing0906 committed:
support arbitary LUT for Nx1 blocks or 8x8 blocks automatically (#15504)

troublemaker52025 committed:
fix csm ubo data in custom pipeline (#15502)

arsen2010 committed:
optimize inspector interface (#15503)

* fix sprite-frame edit button visible bug

* optimize particle-system interface indent

* optimize fbx animation tab header color

* revert

* optimize align

* add debug log error

doommm committed:
fix: cannot find 'lodash' in production build (#15501)

doommm committed:
fix: require path before push module path (#15509)

shrinktofit committed:
Animation: fix PoseGraphNodeInputBinding.outputIndex serialization (#15508)

zxx43 committed:
compatible with 2d (#15505)

* compatible with 2d

* use warn

arsen2010 committed:
optimize interface component header margin right (#15512)

PPpro committed:
fix: add WASM_FALLBACK for spine wasm module (#15511)

* fix: add WASM_FALLBACK for spine wasm module

* update external

* fix eslint

LinYunMo committed:
fix bmfont render bug (#15510)

* fix bmfont render bug

* fix

SantyWang committed:
add parameter binding

dumganhar committed:
Fix spine.wasm could not run in safari with iOS which version is lower than 15. (#15515)

* Fix spine.wasm could not run in safari with iOS which version is lower than 15.

* Update external config.

* Revert "fix: add WASM_FALLBACK for spine wasm module (#15511)"

This reverts commit e5dde7f37d141123a0aac9ac2aaa2764fad27f94.

SantyWang committed:
remove optimization and add compile result part one

SantyWang committed:
rename to acceleration-froce

bofeng-song committed:
fix fsUtils is not defined at taobao platform (#15522)

* fix fsUtils is not defined at taobao platform

PPpro committed:
fix: loading spine module on OPPO using separate engine (#15536)

* fix: loading spine module on OPPO using separate engine

GengineJS committed:
[Fixed] To resolve the abnormal display caused by "FRAMEBUFFER_INCOMPLETE_DIMENSIONS" (#15539)

GengineJS committed:
[Fixed]  Modify instancing cleanup logic. (#15544)

lealzhan committed:
update native/external version v3.8.0-11 (#15546)

shrinktofit committed:
Fix animation clip frame event dispatching (#15534)

SantyWang committed:
add compiling failed operation

troublemaker52025 committed:
Optimize light source luminous_flux and luminance UI step (#15555)

naturalch committed:
update the screen size of multiple LODs may result in an error (#15543)

SantyWang committed:
remove remap of noise

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: f10b26a5acfa5fbc3070b08e5653a3af69f06f4a  (#15555) (#15556)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

novain1102 committed:
revert partial useless code related wechat blackscreen repair (#15554)

PatriceJiang committed:
Filp the y-axis value for stick (#15570)

PPpro committed:
fix: wrong asm.js mem url on minigame platforms (#15559)

* fix: wrong asm.js mem url on Xiaomi platform

* revert binaryUrl fix on OPPO platform

arsen2010 committed:
fix unlink prefab bug in inspector (#15518)

* fix unlink prefab button visible bug in inspector

* fix multiple unlink prefab nodes，record undo is incorrect

* unlink prefab undo has recorded in scene process

* optimize code

2youyou2 committed:
refine post process and add tips (#15566)

arsen2010 committed:
fix edit material cache editing data bug in inspector (#15550)

* optimize edit material in inspector

* update

* optimize multiple selections warning label style

arsen2010 committed:
optimize ui-prop group interface in inspector (#15542)

star-e committed:
V3.8.0 pipeline api document (#15529)

star-e committed:
fix dependency on camera component (#15577)

* fix dependency on camera component

* fix problem caused by removed deprecation

AFeiYA committed:
update migrated meta files (#15581)

jk20012001 committed:
add tooltips for bloom hbao and lut (#15579)

LinYunMo committed:
fix label limit canvas size bug (#15514)

* fix label limit canvas size bug

* limit recursion time

* fix

* fix

* move

* change

* change

* name change

arsen2010 committed:
inspector interface: reserve tab style of group; support new section style; (#15586)

* reserve tab style of group; support new section style;

* add mesh-renderer section group style

yufengctbu committed:
fix inspector document link address (#15521)

lealzhan committed:
[fix] box2d shape aabb calculating bug (#15565)

lealzhan committed:
Revert "[fix] refactor physics2d contact listener (#14026)" (#15564)

* Revert "[fix] refactor physics2d contact listener (#14026)"

* tweaks

jk20012001 committed:
add help for post-process components (#15589)

zhakesi committed:
fix some bugs in spine webassembly (#15569)

* fix blend enum export

* fix convert string from ts to sp string

* update skeleton data in preload

* feature: support change slot texture web

* feature: spine change slot texture native

* fix set debug render in native

* fix some spine asset with no mesh data

* fix some interfaces

* fix change slot repeat to create middleware textures

* fix code style

* update external config

Canvasfull committed:
Fix spine replacement skin and animation bone rendering data not refreshed. (#15597)

SantyWang committed:
Update external-config.json (#15598)

update spine wasm

LinYunMo committed:
Fix widget bug in the editor when scale is not 1 (#15602)

* Fix widget bug in the editor when scale is not 1

* fix scale to 0

* change to approx

LinYunMo committed:
fix label Precision (#15594)

jk20012001 committed:
fix skin diffuse with forward lighting (#15605)

shrinktofit committed:
Animation controller component help (#15609)

jk20012001 committed:
optimize shader code size for include (#15614)

SantyWang committed:
fix performance degradation when redefine the property of renderer's prototype (#15615)

bofeng-song committed:
alipay: the accessSync interface needs to check if it is successful. (#15616)

2youyou2 committed:
fix shadow issue for camera use post process with custom pipeline (#15620)

lealzhan committed:
character controller api tweaks (#15621)

* character controller api tweaks

* tweak

SantyWang committed:
refine unit-test

PPpro committed:
optimize: ensure wasm module ready before importing (#15631)

* optimize: ensure wasm module ready before importing

* Update @types/pal/minigame.d.ts

* Update pal/wasm/wasm-minigame.ts

* use reject instead of warnning

* optimize: add promiseToLoadWasmModule

* add shouldUseWasmModule method to simplify code

* fix: subpackage loading on taobao minigame platform

* Update pal/wasm/wasm-minigame.ts

bofeng-song committed:
Fix the problem that the sound cannot be stopped after calling minigamePlayer.stop on the Alipay platform. (#15633)

* fix audio can not be stopped while call MiniGamePlayer's stop

xubing0906 committed:
fix crash on color grading with GLES (#15634)

troublemaker52025 committed:
fix csm instancedBuffer large primitives count for custom-pipeline (#15636)

LinYunMo committed:
fix spine cache mode bug when no animation (#15637)

naturalch committed:
optimize the text when fbx do not exist lod (#15643)

SantyWang committed:
fix spine artifact in editor (#15640)

dumganhar committed:
Update external to v3.8.0-14 (#15635)

LinYunMo committed:
add renderRoot2d help (#15644)

jk20012001 committed:
fix leaf scattering code (#15645)

jk20012001 committed:
fix shader compilation error (#15646)

star-e committed:
fix native pipeline instancing implementation (#15647)

star-e committed:
remove TextDecoder (#15648)

Canvasfull committed:
Spine v3.8.0 Fix missing interface bindings (#15618)

* Fix spine replacement skin and animation bone rendering data not refreshed.

# Conflicts:
#	cocos/animation/marionette/graph-eval.ts
#	cocos/physics/framework/components/constraints/hinge-constraint.ts
#	cocos/rendering/custom/pipeline.ts
#	cocos/rendering/custom/private.ts
#	cocos/rendering/custom/render-graph.ts
#	cocos/rendering/post-process/passes/forward-pass.ts
#	cocos/rendering/post-process/passes/forward-transparency-pass.ts
#	cocos/rendering/post-process/passes/index.ts
#	cocos/rendering/post-process/passes/tone-mapping-pass.ts
#	cocos/rendering/post-process/post-process-builder.ts
#	editor/assets/effects/advanced/leaf.effect
#	editor/i18n/en/animation.js
#	editor/i18n/en/localization.js
#	editor/i18n/zh/animation.js
#	editor/i18n/zh/localization.js
#	editor/inspector/assets/fbx/model.js
#	native/cocos/renderer/pipeline/custom/NativeExecutor.cpp
#	native/cocos/renderer/pipeline/custom/NativePipeline.cpp
#	native/cocos/renderer/pipeline/custom/NativePipelineTypes.h
#	native/cocos/renderer/pipeline/custom/NativeRenderGraph.cpp
#	native/cocos/renderer/pipeline/custom/RenderInterfaceTypes.h

* Add bone & IkConstraintData & IkConstraint & PathConstraintData & PathConstraint setter bindings.

* Add TrackEntry & AnimationState & Animation setter bindings.

* Fix the problem that SkeletonData removal skin  and animation are not removed on editor.

* fix the problem of skeleton animation delete refresh.

* add Skeleton getState interface.

* optimize the requestDrawData parameter name & fix get texture by textureID.

* add getAttachments

* Add slots and atlas mapping hash.

* optimize atlas hash code.

* parsing atlas from native layer.

* rollback to remove the skeleton component of forced refresh.

* revert some change

* optimize get atlas page index & fix multi-posting texture cache mode does not display.

* fix extra space.

---------

Co-authored-by: zxx43 <zxx43@hotmail.com>
Co-authored-by: santy-wang <wh19930829@qq.com>

PatriceJiang committed:
Update external-config.json (#15649)

Canvasfull committed:
fix spine animation does not render in cache mode when total time is 0. (#15662)

* fix spine animation does not render in cache mode when total time is 0.

* fix code format.

SantyWang committed:
export getAttachments (#15661)

* export getAttachments

* export getAttachments and other interface

* change string to const string&

PatriceJiang committed:
fix CI vulkan-sdk CDN? issue, run vcpkg when choco fails (#15667)

Canvasfull committed:
fix spine  STRING_STD2SP function Uncaught RuntimeError: memory access out of bounds. (#15663)

Co-authored-by: Santy-Wang <wh19930829@qq.com>

Canvasfull committed:
Fix the enumeration order of event types in Wasm (native) and TypeScript interfaces. (#15664)

* Fix the issue of aligning the enumeration order of event types in Wasm (native) and TypeScript interfaces.

* optimize EventType_Complete order.

* fix mapping bindings.

SantyWang committed:
update external version (#15675)

SantyWang committed:
fix problem when skin changed and slot texture changed (#15677)

* fix problem when skin changed and slot texture changed

* update external version

SantyWang committed:
remove useless code (#15682)

SantyWang committed:
Revert "remove useless code (#15682)" (#15683)

This reverts commit 9f035047c7a032031c254462fc87eb09e796271b.

GengineJS committed:
[Remove] Remove the discard check of the managed resource (#15685)

troublemaker52025 committed:
Optimize effect dependency and package size (#15650)

star-e committed:
fix effect.bin serialization (#15690)

SantyWang committed:
Merge branch 'v3.8.0' of github.com:cocos-creator/engine into v3.6.2-particle

SantyWang committed:
add vfx system
